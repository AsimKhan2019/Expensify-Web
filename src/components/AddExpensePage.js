import React from 'react';
import ReactDom from 'react-dom';

const AddExpensePage = () => (
  <div>
    This is from my add expense page component
  </div>
);

export default AddExpensePage;
