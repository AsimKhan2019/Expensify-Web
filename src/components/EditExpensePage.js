import React from 'react';
import ReactDom from 'react-dom';

const EditExpensePage = () => (
  <div>
    This is from my edit expense page component
  </div>
);

export default EditExpensePage;
