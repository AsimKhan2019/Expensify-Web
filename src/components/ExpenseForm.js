import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

export default class ExpenseForm extends React.Component {
  state={
    description: '',
    note: '',
    amount: '',
    createdAt: moment(),
    calendarFocused: false
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState( () => ({description}));
  }

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState( () => ({note}));
  }

  onAmountChange = (e) => {
    const amount = e.target.value;
    if( !amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
      this.setState( () => ({amount}));
    }
  }

  onDateChange = (createdAt) => {
    this.setState( () => ({createdAt}));
  }

  onCalendarFocusedChange = ({focused}) => {
    this.setState( () => ({calendarFocused: focused}));
  }

  render(){
    return(
      <div>
        <form>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}

          />
          <input
            type="text"
            placeholder="Amount"
            autoFocus
            value={this.state.amount}
            onChange={this.onAmountChange}
          />

          <SingleDatePicker
            date={this.state.createdAt} // momentPropTypes.momentObj or null
            onDateChange={this.onDateChange} // PropTypes.func.isRequired
            focused={this.state.calendarFocused} // PropTypes.bool
            onFocusChange={this.onCalendarFocusedChange} // PropTypes.func.isRequired
            numberOfMonths={1}
            isOutsideRange={() => false}
          />


          <textarea
            placeholder="Add a note for your expense (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          >

          </textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
