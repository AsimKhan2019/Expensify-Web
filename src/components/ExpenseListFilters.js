import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate, setCategoryFilter, setPaymentMethodFilter } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

import categories from "../config/categories";
import payment_methods from "../config/payment_methods";


export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onSelectChange = (e) => {
    e.target.value == 'date' ? this.props.sortByDate() : this.props.sortByAmount();
  }
  onCategoryChange = (e) => {
    const v = e.target.value;
    this.props.setCategoryFilter(v);
  }
  onPaymentMethodChange = (e) => {
    const v = e.target.value;
    this.props.setPaymentMethodFilter(v);
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  render() {
    const { dictionary } = this.props;
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input placeholder={dictionary.textFilter} className="text-input" type="text" value={this.props.filters.text} onChange={this.onTextChange} />
          </div>
          <div className="input-group__item">
            <select className="select" value={this.props.filters.sortBy} onChange={this.onSelectChange}>
              <option value="date" >{dictionary.sortByDate}</option>
              <option value="amount" >{dictionary.sortByAmount}</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              numberOfMonths={1}
              isOutsideRange={() => false}
              showClearDates={true}
              startDatePlaceholderText={dictionary.startDatePicker}
              endDatePlaceholderText={dictionary.endDatePicker}
            />
          </div>
        </div>

        <div className="input-group">
          <div className="input-group__item input-group__fluid ">
            <select className="select" value={this.props.filters.category} onChange={this.onCategoryChange}>
              <option value="disabled" >{dictionary.filterByCategoryPlaceholder}</option>
              {categories.map(pm => <option value={pm} key={pm}>
                {dictionary.categories[pm]}
              </option>)}
            </select>
          </div>
          <div className="input-group__item input-group__fluid">
            <select className="select" value={this.props.filters.payment_method} onChange={this.onPaymentMethodChange}>
              <option value="disabled" >{dictionary.filterByPaymentMethodPlaceholder}</option>
              {payment_methods.map(pm => <option value={pm} key={pm}>
                {dictionary.payment_methods[pm]}
              </option>)}
            </select>
          </div>
        </div>
      </div>
    );
  }

}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setCategoryFilter: (endDate) => dispatch(setCategoryFilter(endDate)),
  setPaymentMethodFilter: (endDate) => dispatch(setPaymentMethodFilter(endDate)),
})


const mapStateToProps = (state) => ({
  filters: state.filters,
  dictionary: state.lang.dictionary
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
