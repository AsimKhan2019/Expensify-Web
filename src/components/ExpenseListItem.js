import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({dictonary, id, category, payment_method, description, amount, createdAt, locale}) => (
    <Link className="list-item" to={`/edit/${id}`}>
      <div>
        <h3>{description}</h3>
        <h5 className="list-item-category">{dictonary.categories[category]}</h5>
        <span>{moment(createdAt).locale(locale).format('MMMM Do , YYYY')}</span>   
      </div>
      <div>
        <h3 style={{textAlign: 'right'}}>{numeral(amount / 100).format('$0,0.00')}</h3>
        <h5 style={{textAlign: 'right'}} className="list-item-category">{dictonary.payment_methods[payment_method]}</h5>
      </div>
        
    </Link>
);
//
export default ExpenseListItem;
