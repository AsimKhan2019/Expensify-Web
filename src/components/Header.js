import React from 'react';
import {Link} from 'react-router-dom';
import {startLogout} from '../actions/auth';
import {connect} from 'react-redux';

export const Header = ({startLogout, photoURL, displayName}) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard" ><h1>Expensify</h1></Link>
        <div className="header-user">
          <div><img src={`${photoURL}?sz=50`}></img></div>
          <span className="header-user__info">
            <span className="header-user__displayname">{displayName}</span>
            <button className="button button-link" onClick={startLogout}>Log out</button>
          </span>
        </div>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => ({
  photoURL: state.auth.photoURL,
  displayName: state.auth.displayName
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
