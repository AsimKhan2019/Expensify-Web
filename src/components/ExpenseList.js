import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpenseList = ({expenses, dictionary}) => (
  <div className="content-container">
    <div className="list-header">
      <div >{dictionary.tableExpense}</div>
      <div >{dictionary.tableAmount}</div>
    </div>
    <div className="list-body">
      {
        expenses.length == 0 ? (
          <div className="list-item-message">
            <span>{dictionary.noExpenseMessage}</span>
          </div>
        ) : (
          expenses.map( (expense) => {
            return <ExpenseListItem key={expense.id} dictonary={dictionary.categories} {...expense} />
          })
        )
      }
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses.present, state.filters),
    dictionary: state.lang.dictionary
  };
};

export default connect(mapStateToProps)(ExpenseList);
