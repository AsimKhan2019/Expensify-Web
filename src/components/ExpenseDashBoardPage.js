import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSumary from './ExpenseSumary';
import UndoRedo from './UndoRedo';

const ExpenseDashBoardPage = () => (
  <div>
    <ExpenseSumary/>
    <ExpenseListFilters/>
    <ExpenseList/>
    <UndoRedo/>
  </div>
);

export default ExpenseDashBoardPage;
