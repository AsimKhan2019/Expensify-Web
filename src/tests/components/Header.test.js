import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header';
import en from '../../locale/en';


test('should render Header', () => {
  const wrapper = shallow(
    <Header
      startLogout = {() => {}}
      setLanguage = {() => {}}
      photoURL = "http"
      displayName = "Julio"
      dictionary = {en}
      locale = "en"
    />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(
    <Header
      startLogout = {startLogout}
      setLanguage = {() => {}}
      photoURL = "http"
      displayName = "Julio"
      dictionary = {en}
      locale = "en"
    />);
  wrapper.find('button').at(1).simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
