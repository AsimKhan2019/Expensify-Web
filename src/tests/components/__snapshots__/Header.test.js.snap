// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <div
        className="header__user"
      >
        <div
          className="header__language"
        >
          <button
            className="button button-lang"
            onClick={[Function]}
          >
            EN
          </button>
        </div>
        <div>
          <img
            src="http?sz=50"
          />
        </div>
        <span
          className="header__user__info"
        >
          <span
            className="header__user__displayname"
          >
            Julio
          </span>
          <button
            className="button button-link"
            onClick={[Function]}
          >
            Log out
          </button>
        </span>
      </div>
    </div>
  </div>
</header>
`;
